### PR TITLE
Adding bulk simulator with JSON and Avro formats

### PR DESCRIPTION
**Pull request by mescanne**
This does the following:

- Re-generates the protobuf Java code with a newer version
- Adds in JSON and Avro output types for the Simulator
- Adds a few parameters to customize the output
- Adds example script for producing a lot of output at scale

This is to support generating a large quantity of simulated marketdata in Avro for loading into BigQuery (or elsewhere).

### CODE REVIEW

File: simulator/src/main/java/com/google/cloud/App.java [hunk 1]
```diff
@@ -68,23 +78,23 @@ public class App {
...
       parser.parse(argv);
-      if (app.help) {
+      if (args.help) {
         parser.usage();
         System.exit(1);
       }
 
       // Run the simulator
-      app.runSimulator();
+      args.runSimulator();
```
Comment (slilichenko) [inline simulator/src/main/java/com/google/cloud/App.java]: Looks strange to ask arguments to run something. Perhaps renaming App to SimulatorRunner will be a reasonable naming change. Then "runner.runSimulator" will look pretty logical. Don't have to do it, just a suggestion.

File: simulator/src/main/java/com/google/cloud/App.java [hunk 2]
```diff
@@ -177,14 +194,13 @@ Iterator<List<OrderBookEvent>> buildSimulator() {
    */
   MatcherContext buildMatcherContext() throws ParameterException {
     String sessionId = DateTimeFormatter.ofPattern("yyyy-MM-dd.HH:mm").format(LocalDateTime.now());
-    System.out.println("Session id for this simulation: " + sessionId);
```
Comment (slilichenko) [inline simulator/src/main/java/com/google/cloud/App.java:180]: This output is useful for my tests and is referenced in README. Please make it conditional based on the type of the output.

File: simulator/src/main/java/com/google/cloud/JSONOutputConsumer.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+import com.google.cloud.orderbook.model.MarketDepth;
+import com.google.cloud.orderbook.model.MarketDepth.PriceQuantity;
+import com.google.cloud.orderbook.model.OrderBookEvent;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JSONOutputConsumer implements EventConsumer {
```
Comment (slilichenko) [inline simulator/src/main/java/com/google/cloud/JSONOutputConsumer.java:28]: Perhaps using a standard Json library would have resulted in a shorter code. But it's small enough and we can leave it as-is.

**Review by slilichenko — COMMENTED** [review]
A couple of minor suggestions.

File: simulator/src/main/java/com/google/cloud/App.java (hunk 1 above)
Comment (mescanne) [reply simulator/src/main/java/com/google/cloud/App.java]: Agreed -- reverting this back. Not sure why I ended up doing that.

File: simulator/src/main/java/com/google/cloud/App.java (hunk 2 above)
Comment (mescanne) [reply simulator/src/main/java/com/google/cloud/App.java:180]: Added back. I removed that temporarily as part of taking the JSON output -- loading JSON into BigQuery (at a huge scale with gzip'd files) can be slow. Avro is meaningfully faster.

File: simulator/src/main/java/com/google/cloud/JSONOutputConsumer.java (hunk 3 above)
Comment (mescanne) [reply simulator/src/main/java/com/google/cloud/JSONOutputConsumer.java:28]: Quite possibly. I just wanted to get some JSON output working ASAP. Using Protobuf to JSON routines might have been best.

**Comment by mescanne** [comment]
I've addressed two of the comments, but just left the hard coded JSON code to keep it simpler. Avro was the real solution that I needed.

**Review by slilichenko — APPROVED** [review]

**Pull request merged**